static/js/compose.js

```diff
@@ -535,7 +552,7 @@ function validate_stream_message() {
         }
     }
 
-    if (!validate_stream_message_announcement_only(stream_name)) {
+    if (!validate_stream_message_post_policy(stream_name)) {
         return false;
     }
```

Comment:
Just a note on follow-up work -- we have this function, `validate_stream_message_announce`, which hardcodes the stream name "announce".  We should change that to be a `stream_message_post_policy` option in a future PR.